src/components/Menu.jsx

```diff
@@ -12,6 +12,8 @@ const propTypes = {
   renderItemLabel: PropTypes.func,
   value: PropTypes.string,
   onSelect: PropTypes.func,
+  shouldTransferFocus: PropTypes.bool,
+  notifyBeforeFocusWrap: PropTypes.func,
 };
```

Comment:
What I proposed would allow the listener to override the focus change, if it wants to. 
It would be sufficient to check the return type, and if it's a number or null, override, otherwise don't.

I'm not sure if you actually need that to do what you're doing -- it might be enough for the callback to return a truthy or falsy value, in order to accept or cancel the change. I'm fine with that too. The reason I proposed it the way I did is because of the `focusedIndex = null` line you added:

```
          if (shouldTransferFocus) {
            notifyBeforeFocusWrap(event.key);
            focusedIndex = null;
          }
```

If that's needed, I'd rather do a complete inversion of control, and have the callback decide that focusedIndex should be set to null, rather than having Menu do it.